test/httpServer.cpp

```diff
@@ -203,86 +240,80 @@ void HttpServer::OnRequestBasicAuth(mg_connection* conn, http_message* msg) {
+void HttpServer::OnRequestBasicJson(mg_connection* conn, mg_http_message* /*msg*/) {
     std::string response =
             "[\n"
             "  {\n"
             "    \"first_key\": \"first_value\",\n"
             "    \"second_key\": \"second_value\"\n"
             "  }\n"
             "]";
-    std::string headers = "Content-Type: application/json";
-    mg_send_head(conn, 200, response.length(), headers.c_str());
-    mg_send(conn, response.c_str(), response.length());
+    std::string headers = "Content-Type: application/json\r\n";
+    mg_http_reply(conn, 200, headers.c_str(), response.c_str());
 }
 
-void HttpServer::OnRequestHeaderReflect(mg_connection* conn, http_message* msg) {
-    std::string response = "Header reflect " + std::string{msg->method.p, msg->method.len};
+void HttpServer::OnRequestHeaderReflect(mg_connection* conn, mg_http_message* msg) {
+    std::string response = "Header reflect " + std::string{msg->method.ptr, msg->method.len};
     std::string headers;
     bool hasContentTypeHeader = false;
-    for (size_t i = 0; i < sizeof(msg->header_names) / sizeof(mg_str); i++) {
-        if (!msg->header_names[i].p) {
+    for (auto& header : msg->headers) {
+        if (!header.name.ptr) {
             continue;
         }
 
-        std::string name = std::string(msg->header_names[i].p, msg->header_names[i].len);
+        std::string name = std::string(header.name.ptr, header.name.len);
         if (std::string{"Content-Type"} == name) {
             hasContentTypeHeader = true;
         }
 
         if (std::string{"Host"} != name && std::string{"Accept"} != name) {
-            if (!headers.empty()) {
-                headers.append("\r\n");
-            }
-            if (msg->header_values[i].p) {
-                headers.append(name + ": " + std::string(msg->header_values[i].p, msg->header_values[i].len));
+            if (header.value.ptr) {
+                headers.append(name + ": " + std::string(header.value.ptr, header.value.len) + "\r\n");
             }
         }
     }
 
     if (!hasContentTypeHeader) {
-        if (!headers.empty()) {
-            headers.append("\r\n");
-        }
-        headers.append("Content-Type: text/html");
+        headers.append("Content-Type: text/html\r\n");
     }
-    std::cout << "HEADERS: " << headers << '\n';
-    mg_send_head(conn, 200, response.length(), headers.c_str());
-    mg_send(conn, response.c_str(), response.length());
+    mg_http_reply(conn, 200, headers.c_str(), response.c_str());
 }
 
-void HttpServer::OnRequestTempRedirect(mg_connection* conn, http_message* msg) {
+void HttpServer::OnRequestTempRedirect(mg_connection* conn, mg_http_message* msg) {
     // Get the requested target location:
     std::string location;
-    for (size_t i = 0; i < sizeof(msg->header_names) / sizeof(mg_str); i++) {
-        if (!msg->header_names[i].p) {
+    for (auto& header : msg->headers) {
```

Comment:
Please use the explicit type here and is there a reason this iteration var can not be `const type& header`?